xbmc/platform/win32/WIN32Util.cpp

```diff
@@ -1861,3 +1862,13 @@ bool CWIN32Util::IsDriverVersionAtLeast(const std::string& version1, const std::
   }
   return true;
 }
+
+const std::string CWIN32Util::FormatHRESULT(HRESULT hr)
+{
+  using namespace KODI::PLATFORM::WINDOWS;
+
+  const std::string code = FromW(DXGetErrorStringW(hr));
+  WCHAR buff[2048];
+  DXGetErrorDescriptionW(hr, buff, 2048);
+  return std::format("0x{:X} {} ({})", static_cast<uint32_t>(hr), code, FromW(buff));
```

Comment:
I believe `fmt::format` should be able to format `wchar_t` directly without prior conversion. Requires including `fmt/xchar.h`.

Reply:
Maybe, but I thought that since Kodi is built with C++20 now, we'd move from the StringUtils::Format and the fmt library to std::format.

Reply:
btw the cast is for a signed to unsigned conversion. Without it, a lot of errors would be printed as 0x-7xxxxxxx instead of the recognizable 0x8xxxxxxx form in the MS documentation, include files, etc.

Reply:
> Maybe, but I thought that since Kodi is built with C++20 now, we'd move from the StringUtils::Format and the fmt library to std::format.

I would have thought we settle on fmt because it has the most features and integration work is already done. But that's an other discussion and I'm also fine with std::format :slightly_smiling_face: